### PR TITLE
[improve][offload]keep topic/ns set-offload-policies consistent behavior logic

### PR DESCRIPTION
**Pull request by ethqunzhong**
### Motivation
The existing behavior logic of the namespaces/topics set offload-policies is inconsistent.
1. The namespaces set-offload-policies does not support configuring the filesystem because the `bucket` options are set as `required`.
2. cli options are different, and the behavior of supporting unit conversion is inconsistent. eg：

    For example, there are two admin API methods to configure offload policy for `persistent://myprop/clust/ns1/ds1`:
    a. namespaces set-offload-policies in cli commond `set-offload-policies myprop/clust/ns1 -d s3 -r region -b bucket -e 
    endpoint -mbs 32M -rbs 5M -oat 10M -oats 100 -oae 10s -orp tiered-storage-first`

    b. topics set-offload-policies in cli commond `set-offload-policies persistent://myprop/clust/ns1/ds1 -d s3 -r region -b 
    bucket -e endpoint -m 33554432 -rb 5242880 -t 10485760 -ts 100 -dl 10000 -orp tiered-storage-first`

    The command has the same effect, but their command options are different and whether they recognize parameter units 
    is also different. **This can easily confuse users**.

### Modifications
1. Remove unnecessary condition `bucket` `required=true` in namespace#set-offload-policies.
2. Compatible options for namespace/topic and standardized support for unit recognition

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is already covered by supplementary unit-test.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [x] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:  https://github.com/ethqunzhong/pulsar/pull/7

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
Perfect

+1

**Review by tisonkun — APPROVED** [review]
Great!

**Comment by shy-share** [comment]
> Great!

@tisonkun How do you ensure the correctness of the pr? Did you run the code and test it yourself, or can you tell if there is a problem just by looking at the code? I don't quite understand, please explain

**Comment by tisonkun** [comment]
> correctness

This patch contains tests and the code is correct and straightforward - normal review process.

> please explain

Please avoid an imperative tone. I don't have to explain it to you.

**Comment by tisonkun** [comment]
Blocked by https://github.com/apache/pulsar/issues/20661.

@ethqunzhong please keep an eye on that issue before doing repeatedly retry.

**Comment by ethqunzhong** [comment]
> Blocked by #20661.
> 
> @ethqunzhong please keep an eye on that issue before doing repeatedly retry.

got

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java [hunk 1]
```diff
@@ -2289,7 +2291,7 @@ private class SetOffloadPolicies extends CliCommand {
 
         @Parameter(
                 names = {"--offloadAfterThresholdInSeconds", "-oats"},
-                description = "Offload after threshold seconds (eg: 1,5,10)",
+                description = "Offload after threshold seconds (or minutes,hours,days,weeks eg: 100m, 3h, 2d, 5w).",
                 required = false
         )
```
[the remark this answers is not in the document]
Comment (tisonkun) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java:2296]: Looks good. While this patch makes use of existing utils, I suggest we do a follow-up refactor to review that we can replace handmade utils with lib-provided methods. It wouldn't block this patch from merging, IMO.

[the remark this answers is not in the document]
Comment (ethqunzhong) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaces.java:2296]: > 

@JooHyukKim Thank you for your suggestions. It's appreciated that you took the effort to add them in other PR.

**Comment by tisonkun** [comment]
Merging..

Thank you!

**Pull request merged**